app/presenters/filters_presenter.rb

```diff
@@ -20,10 +20,33 @@ def summary_items
   end
 
   def reset_url
-    "#"
+    return nil unless any_filters?
+
+    finder_url_builder.url_except(all_filter_params)
   end
 
 private
 
   attr_reader :facets, :finder_url_builder
+
+  def applied_filters
+    @applied_filters ||= facets.flat_map(&:applied_filters)
+  end
+
+  # Returns a hash of all relevant query parameters for the currently active filters from all
+  # facets, so they can be removed in one go.
+  def all_filter_params
+    applied_filters
+      .map { _1[:query_params] }
+      .reduce do |acc, query_params|
+        acc.deep_merge(query_params) do |_key, old_val, new_val|
+          if old_val.is_a?(Array)
```

Comment:
Do you want to be doing any combining for hash values at all? I notice the `{ value2: "subvalue" }` in the tests. I'm guessing there's a reason not to do them.